### PR TITLE
route_check: Skip route checks if bgp feature is not enabled

### PR DESCRIPTION
**Pull request by anamehra**
If bgp is not enabled, get_frr_routes() gets empty list and route check fails and throws a traceback. Adding check to to skip route checks  bgp feature is disabled. On the Chassis supervisor, bgp may be disabled. 

Signed-off-by: Anand Mehra anamehra@cisco.com

#### What I did
Fixes https://github.com/sonic-net/sonic-buildimage/issues/17400

Fixed the route_check.py traceback error on Chassis Supervisor.

#### How I did it
Added feature bgp check before performing frr route check. This prevents failure when bgp is not enabled on Supervisor.

#### How to verify it
run route_check.py on Supervisor. No traceback should be observed.

#### Previous command output (if the output of a command-line utility has changed)
```
root@sonic:/home/cisco# route_check.py 
Traceback (most recent call last):
  File "/usr/local/bin/route_check.py", line 822, in <module>
    sys.exit(main()[0])
  File "/usr/local/bin/route_check.py", line 809, in main
    ret, res= check_routes()
  File "/usr/local/bin/route_check.py", line 757, in check_routes
    rt_frr_miss = check_frr_pending_routes()
  File "/usr/local/bin/route_check.py", line 547, in check_frr_pending_routes
    frr_routes = get_frr_routes()
  File "/usr/local/bin/route_check.py", line 343, in get_frr_routes
    output = subprocess.check_output('show ip route json', shell=True)
  File "/usr/lib/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'show ip route json' returned non-zero exit status 1.
```
#### New command output (if the output of a command-line utility has changed)
```
root@sonic:/home/cisco# route_check.py 
root@sonic:/home/cisco# 
```

#### Which release branch to backport (provide reason below if selected)

- [x] 202305

### CODE REVIEW

**Comment by anamehra** [comment]
Hi @abdosi , @gechiang , @rlhui , this PR is ready for review. Needed for 202305. Thanks

**Comment by rlhui** [comment]
only an issue from 202305 up

**Comment by deepak-singhal0408** [comment]
@anamehra can you resolve the conflicts? The files have been modified for multi-asic support...

File: tests/route_check_test_data.py [hunk 1]
```diff
@@ -881,5 +892,40 @@
         },
         RET: -1,
     },
-
+    "21": {
+        DESCR: "basic route check, good one, bgp disabled",
+        ARGS: "route_check -m INFO -i 1000",
+        PRE: {
+            APPL_DB: {
```
Comment (deepak-singhal0408) [inline tests/route_check_test_data.py:899]: You need to pass namespace here? Also need to pass    MULTI_ASIC flag and namespace list..
have a look at other testcases..

**Comment by deepak-singhal0408** [comment]
KeyError: 'namespace-list'



Stack trace
self = <tests.route_check_test.TestRouteCheck object at 0x7f04c0d0bf10>
mock_dbs = None, test_num = '21'

    @pytest.mark.parametrize("test_num", TEST_DATA.keys())
    def test_route_check(self, mock_dbs, test_num):
        logger.debug("test_route_check: test_num={}".format(test_num))
        self.init()
        ret = 0
        ct_data = TEST_DATA[test_num]
        set_test_case_data(ct_data)
>       self.run_test(ct_data)

tests/route_check_test.py:240: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.route_check_test.TestRouteCheck object at 0x7f04c0d0bf10>
ct_data = {'Description': 'basic route check, good one, bgp disabled', 'args': 'route_check -m INFO -i 1000', 'pre-value': {0: {...":"oid:0x3000000000023"}': {}}}, 4: {'DEVICE_METADATA': {'localhost': {}}, 'FEATURE': {'bgp': {'state': 'disabled'}}}}}

    def run_test(self, ct_data):
        with patch('sys.argv', ct_data[ARGS].split()), \
>           patch('sonic_py_common.multi_asic.get_namespace_list', return_value= ct_data[NAMESPACE]), \
            patch('sonic_py_common.multi_asic.is_multi_asic', return_value= ct_data[MULTI_ASIC]), \
            patch('route_check.subprocess.check_output', side_effect=lambda *args, **kwargs: self.mock_check_output(ct_data, *args, **kwargs)), \
            patch('route_check.mitigate_installed_not_offloaded_frr_routes', side_effect=lambda *args, **kwargs: None), \
            patch('route_check.load_db_config', side_effect=lambda: init_db_conns(ct_data[NAMESPACE])):
E           KeyError: 'namespace-list'

tests/route_check_test.py:244: KeyError

**Comment by anamehra** [comment]
Hi @deepak-singhal0408 , I am checking on the errors after the conflict merge.

**Comment by anamehra** [comment]
@deepak-singhal0408 , please review. Thanks!

**Comment by abdosi** [comment]
@anamehra : can we skip complete script execution if `bgp` feature is disable and not tied with only FRR related checks.

**Review by abdosi — COMMENTED** [review]
as per above comment.

**Review by abdosi — APPROVED** [review]

**Comment by abdosi** [comment]
@anamehra please create PR for 202305.

**Comment by anamehra** [comment]
> @anamehra please create PR for 202305.

Thanks for merging this @abdosi ! 202305 needs https://github.com/sonic-net/sonic-utilities/pull/3077. I think that is also the reason for cherry pick conflict.
Please also tag both these PRs for 202311
Thanks

**Comment by StormLiangMS** [comment]
@anamehra cherry pick conflict, pls resubmit separate PR to 202305.

**Comment by yxieca** [comment]
@anamehra @deepak-singhal0408 #3075 and #3077 have blocked sonic-utilities repo sub-module advance in sonic-buildimage repo.

**Comment by deepak-singhal0408** [comment]
Hi @anamehra , could you resubmit this PR? It was reverted earlier.. Thanks!

**Comment by deepak-singhal0408** [comment]
Hi @anamehra , we are waiting on this PR.. if you are busy, please let me know, we can create this PR.. Thanks..
+ @saksarav-nokia

**Comment by anamehra** [comment]
> Hi @anamehra , we are waiting on this PR.. if you are busy, please let me know, we can create this PR.. Thanks..
> 
> * @saksarav-nokia

https://github.com/sonic-net/sonic-utilities/pull/3270
Please review.

**Pull request merged**